### DISCUSSION TITLE
NextJS15 + Styled Components className Hydration error

### DISCUSSION THREAD

**Discussion by mr-greg** [vercel/next.js#73953] [Help]
### Summary

Hello !

In the repo I linked to this question, I have 2 different components with exactly the same content. Only difference is the name of the components.

When rendering the Example component, there is no hydration error.
But when I render the Header component, there is an hydration error for some reason.

Does anyone know why ?
I encoutered this error while trying to reproduce the hydration error coming from the random names generated by styled component from my own project, but ended up doing an even worse monstrosity lol.

Thanks for reading !
(really hope I made a really stupid mistake somewhere and someone will point it out)

### Example

https://github.com/mr-greg/hydration-error-styled-component

**Comment by icyJoseph** [comment]
🤔 

```jsx
// const { default: styled } = require('styled-components');

import styled from 'styled-components';

```

That change fixes it... but why... 

I could guess that the bundler, `use client`, does some weird stuff with the `require`?

**Reply by icyJoseph** [reply]
This is really weird though, if `Header` imports from example, as is, without changes, it works... what?

**Reply by mr-greg** [reply]
Also, if you display both of them with the Example first, you don't have an error, but if you display both with the Header then the Example in the page, the hydration error is there haha

**Reply by icyJoseph** [reply]
I think there's enough grounds to open an issue.  There's a reproduction repository, and no obvious oversight.

**Reply by mr-greg** [reply]
I think so too, I mentioned you in the issue I opened with your solutions as well, thanks for your help !
https://github.com/vercel/next.js/issues/73976

---
nevermind, have to make another project with the template etc, will do that once i get the time
